rumqttc/src/mqttbytes/topic.rs

```diff
@@ -16,8 +37,9 @@ pub fn valid_topic(topic: &str) -> bool {
 /// Checks if the filter is valid
 ///
 /// <https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718106>
-pub fn valid_filter(filter: &str) -> bool {
-    if filter.is_empty() {
+pub fn valid_filter(filter: impl AsRef<str>) -> bool {
```

Comment:
I'd name this `is_filter_valid`

Reply:
Since it is a public function, we may run into the risk of backwards incompatibility due to name change. What do you think?